File: test_auth.py
#!/usr/bin/env python3
"""
Script kiểm tra authentication và token
"""

import requests
import json

# API endpoints
BASE_URL = 'http://localhost:8000/api'
LOGIN_URL = f'{BASE_URL}/users/login/'
STUDENTS_URL = f'{BASE_URL}/database/students/'
TEACHERS_URL = f'{BASE_URL}/database/teachers/'
SUBJECTS_URL = f'{BASE_URL}/database/subjects/'

def test_login():
    """Test login và lấy token"""
    print("🔐 Testing login...")
    
    # Login data
    login_data = {
        'code': '123456789012',  # admin teacher_code
        'password': 'Admin@1234'
    }
    
    try:
        response = requests.post(LOGIN_URL, json=login_data)
        print(f"Status Code: {response.status_code}")
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Login successful!")
            print(f"Access Token: {data.get('access', 'N/A')[:50]}...")
            print(f"Refresh Token: {data.get('refresh', 'N/A')[:50]}...")
            print(f"Role: {data.get('role', 'N/A')}")
            return data.get('access')
        else:
            print(f"❌ Login failed: {response.text}")
            return None
            
    except Exception as e:
        print(f"❌ Error during login: {str(e)}")
        return None

def test_api_with_token(token, endpoint, method='GET', data=None):
    """Test API với token"""
    headers = {
        'Authorization': f'Bearer {token}',
        'Content-Type': 'application/json'
    }
    
    try:
        if method == 'GET':
            response = requests.get(endpoint, headers=headers)
        elif method == 'POST':
            response = requests.post(endpoint, headers=headers, json=data)
        elif method == 'PUT':
            response = requests.put(endpoint, headers=headers, json=data)
        elif method == 'DELETE':
            response = requests.delete(endpoint, headers=headers)
        
        print(f"{method} {endpoint}")
        print(f"Status: {response.status_code}")
        
        if response.status_code in [200, 201, 204]:
            print("✅ Success!")
            if response.content:
                try:
                    result = response.json()
                    print(f"Response: {json.dumps(result, indent=2, ensure_ascii=False)}")
                except:
                    print(f"Response: {response.text}")
        else:
            print(f"❌ Failed: {response.text}")
            
        return response.status_code
        
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        return None

def test_api_without_token(endpoint, method='GET'):
    """Test API không có token"""
    try:
        if method == 'GET':
            response = requests.get(endpoint)
        elif method == 'POST':
            response = requests.post(endpoint)
        elif method == 'DELETE':
            response = requests.delete(endpoint)
        
        print(f"{method} {endpoint} (no token)")
        print(f"Status: {response.status_code}")
        
        if response.status_code in [200, 201, 204]:
            print("✅ Success (no auth required)")
        else:
            print(f"❌ Failed: {response.text}")
            
        return response.status_code
        
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        return None

def main():
    print("🚀 Authentication Test")
    print("=" * 50)
    
    # Test login
    token = test_login()
    
    if not token:
        print("❌ Cannot proceed without token")
        return
    
    print("\n" + "=" * 50)
    print("🔍 Testing API endpoints...")
    
    # Test GET endpoints (should work without token)
    print("\n📋 Testing GET endpoints (no auth required):")
    test_api_without_token(STUDENTS_URL)
    test_api_without_token(TEACHERS_URL)
    test_api_without_token(SUBJECTS_URL)
    
    # Test GET endpoints with token
    print("\n🔐 Testing GET endpoints (with token):")
    test_api_with_token(token, STUDENTS_URL)
    test_api_with_token(token, TEACHERS_URL)
    test_api_with_token(token, SUBJECTS_URL)
    
    # Test POST endpoints (require auth)
    print("\n➕ Testing POST endpoints (require auth):")
    
    # Test create student
    student_data = {
        'student_code': 'TEST001',
        'name': 'Test Student',
        'student_class': 'TEST01',
        'email': 'test@example.com'
    }
    test_api_with_token(token, f'{STUDENTS_URL}create/', 'POST', student_data)
    
    # Test create teacher
    teacher_data = {
        'teacher_code': '123456789013',
        'name': 'Test Teacher',
        'department': 'Test Department',
        'email': 'teacher@example.com'
    }
    test_api_with_token(token, f'{TEACHERS_URL}create/', 'POST', teacher_data)
    
    # Test create subject
    subject_data = {
        'subject_code': 'TEST001',
        'name': 'Test Subject',
        'teacher': 1,  # Assuming teacher ID 1 exists
        'time': 'Thứ 2, 7:00-9:00'
    }
    test_api_with_token(token, f'{SUBJECTS_URL}create/', 'POST', subject_data)
    
    print("\n✅ Authentication test completed!")

if __name__ == "__main__":
    main() 

File: FrontEnd/admin/test_auth.js
// Test authentication script for admin panel
console.log('🔍 Testing authentication...');

// Check if token exists
const token = localStorage.getItem('accessToken');
console.log('Token exists:', !!token);
if (token) {
    console.log('Token preview:', token.substring(0, 50) + '...');
}

// Test API endpoints with authentication
async function testAuth() {
    const API_BASE_URL = 'http://localhost:8000/api';

    // Test 1: GET students (should work without auth)
    console.log('\n📋 Test 1: GET students (no auth required)');
    try {
        const response = await fetch(`${API_BASE_URL}/database/students/`);
        console.log('Status:', response.status);
        if (response.ok) {
            const data = await response.json();
            console.log('✅ Success - Found', data.length, 'students');
        } else {
            console.log('❌ Failed:', response.statusText);
        }
    } catch (error) {
        console.log('❌ Error:', error.message);
    }

    // Test 2: POST create student (requires auth)
    console.log('\n👨‍🎓 Test 2: POST create student (requires auth)');
    if (!token) {
        console.log('❌ No token available');
        return;
    }

    try {
        const studentData = {
            student_code: "123456789012",
            name: "Test Student",
            student_class: "TEST-K99",
            dayofbirth: "01/01/2000",
            email: "test@example.com",
            phone_number: "0123456789",
            address: "Test Address"
        };

        const response = await fetch(`${API_BASE_URL}/database/students/create/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${token}`
            },
            body: JSON.stringify(studentData)
        });

        console.log('Status:', response.status);
        const data = await response.json();

        if (response.ok) {
            console.log('✅ Success - Student created');
            console.log('Student ID:', data.id);
        } else {
            console.log('❌ Failed:', data.message || response.statusText);
        }
    } catch (error) {
        console.log('❌ Error:', error.message);
    }

    // Test 3: Login to get fresh token
    console.log('\n🔐 Test 3: Login to get fresh token');
    try {
        const loginData = {
            code: "079205011306",
            password: "Admin@123"
        };

        const response = await fetch(`${API_BASE_URL}/users/login/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(loginData)
        });

        console.log('Login Status:', response.status);
        const data = await response.json();

        if (response.ok) {
            console.log('✅ Login successful');
            console.log('New token:', data.access.substring(0, 50) + '...');

            // Update token in localStorage
            localStorage.setItem('accessToken', data.access);
            console.log('✅ Token updated in localStorage');
        } else {
            console.log('❌ Login failed:', data.message || response.statusText);
        }
    } catch (error) {
        console.log('❌ Login error:', error.message);
    }
}

// Run tests
testAuth(); 